File: 0020-valid-parentheses/0020-valid-parentheses.c
bool isValid(char* s) {
    int len = strlen(s);
    if (len % 2 != 0) return false;

    for (int i = 0; i < len; i++) {
        if ((s[i] == '{' && s[i+1] == '}') || 
            (s[i] == '[' && s[i+1] == ']') || 
            (s[i] == '(' && s[i+1] == ')')) {
            memmove(&s[i], &s[i + 2], len - i - 1);
            s[len - 2] = '\0';
            len -= 2;
            i = -1; 
        }
    }

   
    return len == 0;
}

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        char[] arr = s.toCharArray();
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<arr.length;i++){
            if(arr[i]=='(' || arr[i]=='[' || arr[i]=='{'){
                stack.push(arr[i]);
            }
            else{
                if(stack.isEmpty()) return false;
                else if(stack.peek()=='(' && arr[i]==')' || 
                        stack.peek()=='[' && arr[i]==']' || 
                        stack.peek()=='{' && arr[i]=='}')
                stack.pop();
                else return false;
            }
        }
        return stack.isEmpty();
    }
}